integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/predicate/MatchIdWithConverterSearchPredicateIT.java

```diff
@@ -0,0 +1,276 @@
+		StubMappingSearchTarget searchTarget = indexManager.createSearchTarget();
+
+		SearchQuery<DocumentReference> query = searchTarget.query()
+				.asReference()
+				.predicate( f -> f.id()
+						.matching( 2 )
+						.matchingAny( Arrays.asList( 1 ) )
+						.toPredicate() )
+				.build();
+
+		assertThat( query )
+				.hasDocRefHitsAnyOrder( INDEX_NAME, DOCUMENT_1, DOCUMENT_2 );
+	}
+
+	@Test
+	public void match_any_single_id() {
+		StubMappingSearchTarget searchTarget = indexManager.createSearchTarget();
+
+		SearchQuery<DocumentReference> query = searchTarget.query()
+				.asReference()
+				.predicate( f -> f.id()
+						.matchingAny( Arrays.asList( 1 ) )
+						.toPredicate() )
+				.build();
+
+		assertThat( query )
+				.hasDocRefHitsAnyOrder( INDEX_NAME, DOCUMENT_1 );
+	}
+
+	@Test
+	public void match_any_ids() {
+		StubMappingSearchTarget searchTarget = indexManager.createSearchTarget();
+
+		SearchQuery<DocumentReference> query = searchTarget.query()
+				.asReference()
+				.predicate( f -> f.id()
+						.matchingAny( Arrays.asList( 1, 3 ) )
+						.toPredicate() )
+				.build();
+
+		assertThat( query )
+				.hasDocRefHitsAnyOrder( INDEX_NAME, DOCUMENT_1, DOCUMENT_3 );
+	}
+
+	private void initData() {
+		IndexWorkPlan<? extends DocumentElement> workPlan = indexManager.createWorkPlan();
+		workPlan.add( referenceProvider( DOCUMENT_1 ), document -> {
+			indexMapping.supportedFieldModels.forEach( f -> f.document1Value.write( document ) );
+			indexMapping.supportedFieldWithDslConverterModels.forEach( f -> f.document1Value.write( document ) );
+			indexMapping.unsupportedFieldModels.forEach( f -> f.document1Value.write( document ) );
+		} );
+		workPlan.add( referenceProvider( DOCUMENT_2 ), document -> {
+			indexMapping.supportedFieldModels.forEach( f -> f.document2Value.write( document ) );
+			indexMapping.supportedFieldWithDslConverterModels.forEach( f -> f.document2Value.write( document ) );
+			indexMapping.unsupportedFieldModels.forEach( f -> f.document2Value.write( document ) );
+		} );
+		workPlan.add( referenceProvider( DOCUMENT_3 ), document -> { } );
+		workPlan.add( referenceProvider( EMPTY ), document -> { } );
+		workPlan.execute().join();
+
+		// Check that all documents are searchable
+		StubMappingSearchTarget searchTarget = indexManager.createSearchTarget();
+		SearchQuery<DocumentReference> query = searchTarget.query()
+				.asReference()
+				.predicate( f -> f.matchAll().toPredicate() )
+				.build();
+		assertThat( query ).hasDocRefHitsAnyOrder( INDEX_NAME, DOCUMENT_1, DOCUMENT_2, EMPTY,
+				DOCUMENT_3
+		);
+	}
+
+	private static class IndexMapping {
+		final List<ByTypeFieldModel<?>> supportedFieldModels;
+		final List<ByTypeFieldModel<?>> supportedFieldWithDslConverterModels;
+		final List<ByTypeFieldModel<?>> unsupportedFieldModels;
+
+		IndexMapping(IndexSchemaElement root) {
+			supportedFieldModels = mapSupportedFields( root, "supported_", ignored -> { } );
```

Comment:
Note to myself: this is unnecessarily complex, as is the setup in `MatchIdSearchPredicateIT`. I'll try to simplify it.